operators/tokenizer/bpe_kernels.cc

```diff
@@ -641,3 +643,147 @@ OrtxStatus JsonFastTokenizer::Compute(const ortc::Tensor<std::string>& input,
                                       std::optional<ortc::Tensor<int64_t>*> offset_mapping) const {
   return KernelBpeTokenizer::Compute(input, tokenize_output, attention_mask, offset_mapping);
 }
+
+TikTokenizer::TikTokenizer() : KernelBpeTokenizer(kGPT2Configuration) {}
+
+std::string TikTokenizer::TokenBytesToString(std::vector<uint8_t>& bytes) {
+    std::string result;
+    for (auto c : bytes) {
+        result += bbpe_tokenizer_->ByteEncoder()[static_cast<unsigned char>(c)];
+    }
+    return result;
+}
+
+// Custom hash function for the vector key
+struct VectorHash {
+    size_t operator()(const std::vector<uint8_t>& v) const {
+        std::hash<uint8_t> hasher;
+        size_t seed = 0;
+        for (uint8_t i : v) {
+            seed ^= hasher(i) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+        }
+        return seed;
+    }
+};
+
+// Custom equality function for the vector key
+struct VectorEqual {
+    bool operator()(const std::vector<uint8_t>& a, const std::vector<uint8_t>& b) const {
+        return a == b;
+    }
+};
+
+OrtxStatus TikTokenizer::Load(const ort_extensions::bpe::TokenJsonConfig& config) {
+  std::string voc_file = config.GetVocabDataFile();
+  std::ifstream ifs = path(voc_file).open();
+  if (!ifs.is_open()) {
+    return OrtxStatus(kOrtxErrorInvalidFile, "Failed to open vocab file: " + voc_file);
+  }
+
+  std::unordered_map<std::vector<uint8_t>, uint32_t, VectorHash, VectorEqual> bpe_ranks;
```

Comment:
why not directly use std::string instead of std::vector<uint8_t>?
std::string also stores the length of the string so you don't need to worry about '\x0' char in the string

Reply:
because we are decoding the strings in base64 in the .tiktoken file from std::string to std::vector<uint8_t> first - let me know if you think this thinking is correct.

Reply:
you can convert std::vector<int8_t> into a std::string, so that we don't need these map function and a simpler type define,

Reply:
another (and more crucial) reason that the bpe_ranks need to have the vector of bytes is because those bytes get encoded using our ByteEncoder first; after, actually the vocab map here: https://github.com/microsoft/onnxruntime-extensions/pull/729/files#diff-525f03c6db82461d5659425f865848836162b44f0559d7338f11a93f9c3f28bdR701 uses strings - so the string conversion does happen for simpler types, just later in the primitive.